CRM/Sepa/BAO/SEPAMandate.php

```diff
@@ -49,7 +48,11 @@ static function add(&$params) {
     if (empty($params['id'])) {
       CRM_Utils_SepaCustomisationHooks::create_mandate($params);
 
-      if (empty($params['reference'])) {
+      if (empty($params['reference']) && !empty($params['creditor_id'])) {
+        $creditor = civicrm_api3('SepaCreditor', 'getsingle', [
+          'id' => $params['creditor_id'],
+          'return' => 'mandate_prefix,creditor_type'
+        ]);
         // If no mandate reference was supplied by the caller nor the customisation hook, create a nice default one.
```

Comment:
Doesn't this mean, that if no creditor is submitted, no reference will be generated? Do you mind if I extend it to using the default creditor in that case?